### ISSUE TITLE
XML::LibXML 2.0201+ requires enabling of External DTDs

### ISSUE THREAD

**Issue by drn05r** [eprints/eprints#511]
Reported by Paolo Tealdi via Eprints-tech

this version the module needs one more MANDATORY option at startup, that enables 
external DTD parsing, now disabled by default.

**Comment by jesusbagpuss** [comment]
Just adding this note - gathering the info in one place.

From EP-Tech:

```
--- XML/LibXML_old.pm 2017-04-28 11:03:16.000000000 +0200
+++ XML/LibXML.pm     2020-01-30 08:42:55.273103000 +0100
@ AT  -66,7 +66,7 @ AT 

 ##############################################################################

-our $PARSER = XML::LibXML->new();
+our $PARSER = XML::LibXML->new( expand_entities=>1, load_external_dtd=>1);

 sub CLONE
 {
```

**Comment by jesusbagpuss** [comment]
Other places that might need to be addressed:

https://github.com/eprints/eprints/blob/2d42b2be54004bf3db6881a6d893d39c7a7f909f/perl_lib/EPrints/XML/LibXML.pm#L71-L74

https://github.com/eprints/eprints/blob/2d42b2be54004bf3db6881a6d893d39c7a7f909f/perl_lib/EPrints/Plugin/Import/XSLT.pm#L51

**Comment by denics** [comment]
Hi guys, as this is of actuality again :) I was wondering few years ago, why EPrints is including these libraries as part of the code and it is not using a dependency system

**Comment by jesusbagpuss** [comment]
@denics it might be because, inolder versions, EPrints supported other XML libraries (GDOME, DOM), and the EPrints::XML::zzz modules were to create a more standard way to interface to all of them?
(but that was all a long time ago - and I might be wrong!)

I'd hope that somewhere in the 3.4 series, there will be a review of the code base, and refactor on a standard library (LibXML seems to be the winner!?).

**Comment by denics** [comment]
Yep, indeed you are right, but now we should be able to get rid of all of them. I do not remember where I did propose some patches, I do not think in here, maybe in the eprintsug group. I remember that I did that because exactly libXML that had a major critical bug and EP was still using the old one. It is not a big job to rely on system libs now Perl using carton or even cpanm.
